Sources/InputTextView.swift

```diff
@@ -0,0 +1,100 @@
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import UIKit
+
+open class InputTextView: UITextView {
+
+    // MARK: - Properties
+
+    open var placeholder: NSString? {
+        didSet { setNeedsDisplay() }
+    }
+
+    open var placeholderColor: UIColor = .lightGray {
+        didSet { setNeedsDisplay() }
+    }
+
+    open var placeholderLeftMargin: CGFloat = 4 {
+        didSet { setNeedsDisplay() }
+    }
+
+    // MARK: - Initializers
+
+    override public init(frame: CGRect, textContainer: NSTextContainer?) {
+        super.init(frame: frame, textContainer: textContainer)
+        commonInit()
+    }
+
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    // MARK: - Methods
+
+    override open func draw(_ rect: CGRect) {
+        super.draw(rect)
+        if text.isEmpty {
+            guard let placeholder = placeholder else { return }
+            let paragraphStyle = NSMutableParagraphStyle()
+            paragraphStyle.alignment = textAlignment
+
+            let rect = CGRect(x: textContainerInset.left + placeholderLeftMargin,
+                              y: textContainerInset.top,
+                              width:   frame.size.width - textContainerInset.left - textContainerInset.right,
+                              height: frame.size.height)
+
+            var attributes: [String: Any] = [
+                NSForegroundColorAttributeName: placeholderColor,
+                NSParagraphStyleAttributeName: paragraphStyle
+            ]
+            if let font = font {
+                attributes[NSFontAttributeName] = font
+            }
+
+            placeholder.draw(in: rect, withAttributes: attributes)
+        }
+    }
+
+    fileprivate func commonInit() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(textDidChange),
+                                               name: NSNotification.Name.UITextViewTextDidChange,
+                                               object: self)
+    }
+
+    func textDidChange(notification: Notification) {
```

Comment:
Any specific reason why we moved from using the delegate to the `NotificationCenter`?